### PR TITLE
Implement MLIR L1 interface on top of graph capture

### PR DESCRIPTION
**Pull request by derdeljanTT**
To all folks who are code owners, no review required from the code owners, this is our internal branch that we are working on TTNN <-> optimiser interface.

Implemented MLIR L1 interface API on top of graph capture. Added feature switch `TTNN_MLIR_INTERFACE_USE_GRAPH_CAPTURE` to enable graph capture mode for L1 interface, by default analytical model is used.

### CODE REVIEW

File: ttnn/cpp/ttnn/mlir_graph_capture_l1_interface.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    const L1InterfaceOperandParams& input_a,
+    const L1InterfaceOperandParams& input_b,
+    const L1InterfaceOperandParams& output) {
+    ScopedDeviceContext ctx;
+
+    auto input_tensor_a = create_tensor(ctx.get_device(), input_a);
+    auto input_tensor_b = create_tensor(ctx.get_device(), input_b);
+
+    auto call = [&] {
+        const auto output_tensor = ttnn::add(
```
Comment (mbezuljTT) [inline ttnn/cpp/ttnn/mlir_graph_capture_l1_interface.cpp:70]: is it feasible to (long term) replace ttnn:add with string eltwise_op_name or something like that? not urgent - only to understand this solution better.

**Review by mbezuljTT — APPROVED** [review]

File: ttnn/cpp/ttnn/mlir_graph_capture_l1_interface.cpp (hunk 1 above)
Comment (derdeljanTT) [reply ttnn/cpp/ttnn/mlir_graph_capture_l1_interface.cpp:70]: It is, I will add a TODO in the code

**Comment by davorchap** [comment]
would this interface be useful for other stack integrations , or is it MLIR specific?

**Comment by mbezuljTT** [comment]
> would this interface be useful for other stack integrations , or is it MLIR specific?

Good question; I'll talk with Artem to understand how they plan to use GraphCapture with Torch compiler. 
At the moment we are exploring if we can use it from MLIR, thus we have somewhat heavy wrapping.

**Pull request merged**